api/candlepin-api-spec.yaml

```diff
@@ -1724,6 +1722,127 @@ components:
             upstreamConsumer:
               $ref: '#/components/schemas/UpstreamConsumerDTO'
 
+    PermissionBlueprintDTO:
+      description: Represents the permission blueprints exposed to the API layer
+      allOf:
+        - $ref: '#/components/schemas/TimestampedEntity'
+        - type: object
+          properties:
+            id:
+              type: string
+              example: "ff808081554a3e4101554a3e9033005d"
+            owner:
+              $ref: "#/components/schemas/NestedOwnerDTO"
+            type:
+              type: string
+              example: "OWNER"
+            access:
+              type: string
+              example: "NONE"
+
+    PoolDTO:
+      description: Pool DTO representation
+      required:
+        - productId
```

Comment:
The PoolDTO.productId field is not currently validated upon (the ProvidedProductDTO.productId is, and it rightly has it as required), so this section is not required at all on PoolDTO.

Also, if we were to need required fields, this is the wrong place for this element (it does not result in `@notNull` annotation being generated. it would have to be under allOf: -type: object (at the same level as the properties: element) for it to work. This is something I noticed we did in ContentDTO, and we should fix (move the required: element at the same level as properties:)

Reply:
Done.